AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift

```diff
@@ -14,26 +14,69 @@ class SearchViewModel: ObservableObject {
     @Published var addressName = ""
     @Published var addresslocation = CLLocationCoordinate2D(latitude: 20.0, longitude: 20.0)
     @Published var alert: AlertInfo?
-    @Published private var parameter: MapDownloaderMeasurementType = .ozone
-    @Published private var sensor: MapDownloaderSensorType = .OzoneSensor
+    @Published var measurementTypes = [SearchParameter]()
+    @Published var sensorTypes = [SearchParameter]()
+    
+    init() {
+        measurementTypes = [
+            .init(isSelected: true, name: Strings.SearchFollowParamNames.particulateMatter),
+            .init(isSelected: false, name: Strings.SearchFollowParamNames.ozone)
+        ]
+        onParameterTap(with: Strings.SearchFollowParamNames.particulateMatter)
+    }
     
     var continueDisabled: Bool { addressName == "" }
-    var shoudShowPMChoiceSheet: Bool { parameter == .particulateMatter }
-    var getParameter: MapDownloaderMeasurementType { parameter }
-    var getSensor: MapDownloaderSensorType { sensor }
+    
+    var selectedParameter: MeasurementType? {
+        guard let selected = measurementTypes.first(where: { $0.isSelected }) else { return nil }
+        return measurementType(from: selected.name)
+    }
+    
+    var selectedSensor: SensorType? {
+        guard let selected = sensorTypes.first(where: { $0.isSelected }) else { return nil }
+        return sensorType(from: selected.name)
+    }
+    
     func textFieldTapped() { isLocationPopupPresented.toggle() }
     
-    func onPMSensorTap(with sensor: PMSensorType) {
-        self.sensor = MapDownloaderSensorType.allCases.first(where: { $0.capitalizedName == sensor.capitalizedName }) ?? .AB3and2
+    private func measurementType(from: String) -> MeasurementType? {
+        switch from {
+        case Strings.SearchFollowParamNames.particulateMatter: return .particulateMatter
+        case Strings.SearchFollowParamNames.ozone: return .ozone
+        default: return nil
+        }
+    }
+    
+    private func sensorType(from: String) -> SensorType? {
+        switch from {
+        case Strings.SearchFollowSensorNames.AirBeam3and2: return .AB3and2
+        case Strings.SearchFollowSensorNames.openAQ: return .OpenAQ
+        case Strings.SearchFollowSensorNames.purpleAir: return .PurpleAir
+        case Strings.SearchFollowSensorNames.openAQOzone: return .OpenAQ
+        default: return nil
+        }
     }
     
-    func onOzoneSensorTap(with sensor: OzoneSensorType) {
-        self.sensor = MapDownloaderSensorType.allCases.first(where: { $0.capitalizedName == sensor.capitalizedName }) ?? .OzoneSensor 
+    func onParameterTap(with param: String) {
```

Comment:
a bit messy structure in this file. Please move all `public` methods above `private` ones to improve readability (and do the same with properties)